src/com/linkedin/parseq/internal/SerialExecutor.java

```diff
@@ -117,7 +122,12 @@ public void run() {
       // Entering state:
       // - _queue.size() > 0
       // - _pendingCount.get() > 0
+
+      final ExecutionMonitorState executionState = _executionMonitor != null ? _executionMonitor.getLocalMonitorState() : null;
```

Comment:
there's a lot of null checks, is it possible to make sure that executionMonitor is never null in the constructor (create NoOp executionMonitor instead of null?)